torch/_higher_order_ops/scan.py

```diff
@@ -357,9 +456,176 @@ def scan_op_dense(combine_fn, init, xs, dim, reverse, additional_inputs):
     return generic_scan(combine_fn, init, xs, dim, reverse, additional_inputs)
 
 
-scan_op.py_impl(DispatchKey.Autograd)(
-    autograd_not_implemented(scan_op, deferred_error=True)
-)
+class ScanAutogradOp(torch.autograd.Function):
+    @staticmethod
+    def forward(
+        ctx,
+        fw_graph,
+        joint_graph,
+        dim,
+        reverse,
+        num_leaves_init,
+        num_leaves_additional_inputs,
+        *ops,
+    ):
+        ops_list = list(ops)
+        init = ops_list[:num_leaves_init]
+        xs = ops_list[num_leaves_init : len(ops_list) - num_leaves_additional_inputs]
+        additional_inputs = ops_list[len(ops_list) - num_leaves_additional_inputs :]
```

Comment:
Maybe a locally scoped helper function:
`def split_operands(ops, num_leaves_init, num_leaves_additional_inputs)`

nit: ops usually refers to operators, so probably change it flat_args or something

Reply:
Done, I simplified and unified the code